src/timezones/io.jl

```diff
@@ -0,0 +1,98 @@
+import Base.Dates: Slot, AbstractTime, FixedWidthSlot, DelimitedSlot, DayOfWeekSlot, duplicates, getslot, periodisless
+
+Base.string(tz::TimeZone) = string(tz.name)
+Base.show(io::IO,tz::VariableTimeZone) = print(io,string(tz))
+
+function Base.string(dt::ZonedDateTime)
+    v = offset(dt.zone).value
+    h, v = divrem(v, 3600)
+    m, s  = divrem(abs(v), 60)
+
+    hh = @sprintf("%+03i", h)
+    mm = lpad(m, 2, "0")
+    ss = s != 0 ? lpad(s, 2, "0") : ""
+
+    local_dt = localtime(dt)
+    return "$local_dt$hh:$mm$(ss)"
+end
+Base.show(io::IO,dt::ZonedDateTime) = print(io,string(dt))
+
+
+# DateTime Parsing
+
+# Note: Ideally Base.Dates.Slot definition should just be: "abstract Slot{P<:Any}" which
+# would allow for custom Slots to be of any type.
+immutable TimeZoneSlot <: AbstractTime end
```

Comment:
Yeah, we're really due for another iteration on the Base.Dates io.jl machinery. There are a few open issues about parsing issues and I've just been putting off really digging in (because parsing isn't my favorite activity!). If you have ideas/code here, it'd be much appreciated in Base.